manifests/repository.pp

```diff
@@ -5,10 +5,22 @@
 # This class will be automatically included when a resource is defined.
 # It is not intended to be used directly by external resources like node definitions or other modules.
 class bareos::repository (
-  $release = 'latest',
-  $gpg_key_fingerprint = undef,
+  String           $release             = 'latest',
+  Optional[String] $gpg_key_fingerprint = undef,
+  Boolean          $subscription        = false,
+  String           $username            = absent,
```

Comment:
I think you probably want

```puppet
Optional[String[1]] $username = undef,
Optional[String[1]] $password = undef,
```

Reply:
no the default should be absent, because if no username/pw is set it need to be removed completly from the repo file.
thats done via yumrepo if they are `absent`
https://puppet.com/docs/puppet/5.5/types/yumrepo.html#yumrepo-attribute-password

Reply:
although since the defaults are already set in init.pp we could change this to undef here right?

Reply:
Aha, I see. I might read easier wrt data typing to do something like 

```puppet
if $username == undef {
  $_username = absent
} else {
  $_username = $username
}
```

Reply:
I don't see why thats easier. I would have to add these lines for both username and password which is more complex than just setting the values as default.

Reply:
Well, since I had to change the whole thing again to distinguish between apt and yum I ended up with initialising the credentials with `undef` and then setting them in the repository.pp
Does this suit you?

Reply:
@ghoneycutt are you happy with the changes?